Common/Thread/ThreadManager.cpp

```diff
@@ -98,10 +98,6 @@ static void WorkerThreadFunc(GlobalThreadContext *global, ThreadContext *thread)
 }
 
 void ThreadManager::Init(int numRealCores, int numLogicalCoresPerCpu) {
-	if (!global_->threads_.empty()) {
-		Teardown();
-	}
-
```

Comment:
Does this part actually hurt? Seems that just calling Teardown from retro_unload_game should be enough, without deleting these three lines. I don't remember exactly why I added these though, heh.

Reply:
It seemed redundant in terms of the lifecycle of the thread manager in the main build. As far as i can reason init is only ever called once and any full scale cleanup is handled by exiting the application. As a library it does hang around way longer and the cleanup was needed so decoupling the init and teardown behaviours when we have appropriate lifecycle callbacks seemed logical to me. That teardown call was the source of the issue in the first place so if for whatever reason the new retro_unload_game based teardown call fails we would fall back to the whatever race condition that got us here in the first place.

That said I have not been able to figure out why nothing was rendering. On Nvidia and AMD machines it was 100% reproducable yet on an Intel Iris Xe there were zero issues. Plus I didnt notice any rendering related tasks hitting that thread pool so im at a loss as to what was going on. I'm in a here be dragons part of the map so if you think it needs to come back im not going to argue :)

Reply:
I would prefer that the change is not in this PR unless it's really needed. I'm gonna clean it up myself and test the other platforms separately though.

Reply:
Pushed the revert now.

Reply:
Thanks!